layers/+lang/clojure/packages.el

```diff
@@ -315,12 +272,63 @@
       (add-to-list 'auto-mode-alist '("\\.boot\\'" . clojure-mode))
       ;; This regexp matches shebang expressions like `#!/usr/bin/env boot'
       (add-to-list 'magic-mode-alist '("#!.*boot\\s-*$" . clojure-mode))
-      (dolist (m '(clojure-mode clojurec-mode clojurescript-mode clojurex-mode))
-        (spacemacs/set-leader-keys-for-major-mode m
-          "fl" 'clojure-align)))
+      ;; Define all the prefixes here, although most of them apply only to bindings in clj-refactor
+      (let ((clj-refactor--key-binding-prefixes
+             '(("mr" . "refactor")
+               ("mra" . "add")
+               ("mrc" . "cycle/clean/convert")
+               ("mrd" . "destructure")
+               ("mre" . "extract/expand")
+               ("mrf" . "find/function")
+               ("mrh" . "hotload")
+               ("mri" . "introduce/inline")
+               ("mrm" . "move")
+               ("mrp" . "project/promote")
+               ("mrr" . "remove/rename/replace")
+               ("mrs" . "show/sort/stop")
+               ("mrt" . "thread")
+               ("mru" . "unwind/update"))))
+        (spacemacs|forall-clojure-modes m
+          (mapc (lambda (x) (spacemacs/declare-prefix-for-mode
+                              m (car x) (cdr x)))
+                clj-refactor--key-binding-prefixes)
+          (spacemacs/set-leader-keys-for-major-mode m
+            "fl" 'clojure-align
+            "rci" 'clojure-cycle-if
+            "rcp" 'clojure-cycle-privacy
+            "rc#" 'clojure-convert-collection-to-set
+            "rc'" 'clojure-convert-collection-to-quoted-list
+            "rc(" 'clojure-convert-collection-to-list
+            "rc[" 'clojure-convert-collection-to-vector
+            "rc{" 'clojure-convert-collection-to-map
+            "rc:" 'clojure-toggle-keyword-string
+            "rtf" 'clojure-thread-first-all
+            "rth" 'clojure-thread
+            "rtl" 'clojure-thread-last-all
+            "rua" 'clojure-unwind-all
+            "ruw" 'clojure-unwind)
+          (mapc (lambda (x) (spacemacs/declare-prefix-for-mode
+                              m (car x) (cdr x)))
+                clj-refactor--key-binding-prefixes)
+          (unless clojure-enable-clj-refactor
+            (spacemacs/set-leader-keys-for-major-mode m
+              "r?" 'spacemacs/clj-describe-missing-refactorings)
+            (defun spacemacs/clj-describe-missing-refactorings ()
```

Comment:
This function should be defined in the `funcs.el` file from the same directory.

Reply:
Oh okay, I thought it would reduce overhead and make sense to only define it in the case that clj-refactor was not installed?

Reply:
While that is one benefit of your approach, we don't do that anywhere else in the code and we've refactoed this in the past to move conditionally defined functions in the `funcs.el` file, for better separation of concerns.

Reply:
Yes please squash all commits into one.